multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/variables/BatchToProcessSerializationAdapter.java

```diff
@@ -0,0 +1,83 @@
+package org.cloudfoundry.multiapps.controller.process.variables;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.cloudfoundry.multiapps.common.ParsingException;
+import org.cloudfoundry.multiapps.common.util.JsonUtil;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.core.cf.CloudHandlerFactory;
+import org.cloudfoundry.multiapps.controller.core.cf.v2.ServicesCloudModelBuilder;
+import org.cloudfoundry.multiapps.controller.process.steps.StepsUtil;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.flowable.common.engine.api.variable.VariableContainer;
+import org.flowable.variable.api.delegate.VariableScope;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+/**
+ * 
+ * Needed for backwards compatibility due to changes in how this parameter was serialized between versions - from Resource to a
+ * CloudServiceInstanceExtended. This class should be removed in a following release.
+ *
+ */
+public class BatchToProcessSerializationAdapter implements Serializer<List<CloudServiceInstanceExtended>> {
+
+    public static TypeReference<List<CloudServiceInstanceExtended>> BATCH_TO_PROCESS_LIST_TYPE_REFERENCE = new TypeReference<>() {
+    };
+    public static TypeReference<CloudServiceInstanceExtended> BATCH_TO_PROCESS_ELEMENT_TYPE_REFERENCE = new TypeReference<>() {
+    };
+
+    private static TypeReference<List<Resource>> RESOURCE_LIST_TYPE_REFERENCE = new TypeReference<>() {
+    };
+
+    protected ServicesCloudModelBuilder getServicesCloudModelBuilder(VariableContainer container) {
+        CloudHandlerFactory handlerFactory = StepsUtil.getHandlerFactory((VariableScope) container);
+        DeploymentDescriptor deploymentDescriptor = VariableHandling.get(container, Variables.COMPLETE_DEPLOYMENT_DESCRIPTOR);
+        String namespace = (String) container.getVariable(Variables.MTA_NAMESPACE.getName());
+        return handlerFactory.getServicesCloudModelBuilder(deploymentDescriptor, namespace);
+    }
+
+    @Override
+    public Object serialize(List<CloudServiceInstanceExtended> value) {
+        return value.stream()
+                    .map(JsonUtil::toJson)
+                    .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<CloudServiceInstanceExtended> deserialize(Object serializedValue, VariableContainer container) {
+        List<CloudServiceInstanceExtended> instances = new ArrayList<>();
+        if (serializedValue instanceof String) {
```

Comment:
If the value is String, doesn't this mean the BATCH_TO_PROCESS is set as a resource, because it was set as JsonStringVariable?